pkg/cloudprovider/providers/aws/aws_routes.go

```diff
@@ -64,9 +81,10 @@ func (s *AWSCloud) ListRoutes(clusterName string) ([]*cloudprovider.Route, error
 			continue
 		}
 
-		instance, err := s.getInstanceById(instanceID)
-		if err != nil {
-			return nil, err
+		instance, found := instances[instanceID]
+		if !found {
+			fmt.Errorf("unable to find instance ID %s in the list of instances being routed to", instanceID)
```

Comment:
Oops... missed this one before - this should be `glog.Warningf` or similar, not fmt.Errorf.  fmt.Errorf just returns an error object, which we are then ignoring.